resend/client.go

```diff
@@ -39,14 +39,14 @@ func (r *Resend) SendEmailAsync(ctx context.Context, subject, text, html string,
 			"task": "sending email",
 		},
 		Fn: func(ctx context.Context) error {
-			_, err := r.client.Emails.SendWithContext(ctx, req)
+			_, err := c.client.Emails.SendWithContext(ctx, req)
 			if err != nil {
-				return emailerrors.NewEmailSendingFailedError().Wrap(err)
+				return fmt.Errorf("failed to send email via Resend: %w", err)
```

Comment:
Emails are sent asynchronously, and potential errors are only logged, not handled in the service. Therefore, there is no need to have `pkg/emailerrors`.